core/lib/generators/solidus/update/update_generator.rb

```diff
@@ -0,0 +1,96 @@
+# frozen_string_literal: true
+
+require 'spree/preferences/preference_differentiator'
+require 'rails/generators'
+
+module Solidus
+  # @private
+  class UpdateGenerator < ::Rails::Generators::Base
+    FROM = Spree.previous_solidus_minor_version
+
+    desc 'Generates a new initializer to preview the new defaults for current Solidus version'
+
+    source_root File.expand_path('templates', __dir__)
+
+    class_option :initializer_basename,
+                 type: :string,
+                 default: 'new_solidus_defaults',
+                 banner: 'The name for the new initializer'
+
+    class_option :previous_version_prompt,
+                 type: :boolean,
+                 default: true,
+                 banner: 'Prompt to warn about only previous version support'
+
+    class_option :from,
+                 type: :string,
+                 default: FROM,
+                 banner: 'Solidus version from which you are upgrading'
+
+    class_option :to,
+                 type: :string,
+                 default: Spree.solidus_version,
+                 hide: true
+
+    class_option :initializer_directory,
+                 type: :string,
+                 default: 'config/initializers/',
+                 hide: true
+
+    def create_new_defaults_initializer
+      previous_version_prompt = options[:previous_version_prompt]
+      return if previous_version_prompt && !yes?(<<~MSG, :red)
+        The update process is only supported if you are coming from version #{FROM}. If this is not the case, please, skip it and update your application to use Solidus #{FROM} before retrying.
```

Comment:
What if, instead of having the `previous_version_prompt` option, we always showed the warning, but only when the user passed a `--from` version that's older than the previous minor version?

So if I'm on Solidus 2.11 and run the update generator with no options (or if I run it with `--from=2.10`), I don't get the warning. But if I run it with `--from=2.9`, I get the warning and I can decide if I want to continue or not.

I would also update this message: it doesn't make much sense to tell people that a use case is unsupported but still allow them to continue, imo. We could tell them something like:

> You specified you want to upgrade from Solidus [from version] to Solidus [current version], but the upgrade process is tested to be run one minor release at a time. Instead of skipping versions, we strongly recommend upgrading to Solidus [from version + 1 minor], fixing any deprecations and then continuing from there.
>
> If you are confident you want to upgrade all the way from Solidus [from version] to Solidus [current version] instead, you can still do it, although the process might be more complicated.
> 
> Are you sure you want to continue? (y/N)

Reply:
> What if, instead of having the previous_version_prompt option, we always showed the warning, but only when the user passed a --from version that's older than the previous minor version?

That would mean that in the case the `--from` version is not specified, we'll print no warning. However, it's precisely those users who we want to address the most. The idea was to document a seamless upgrade process to bump from a minor version to the next one and then run `bin/rails g solidus:update`. If a user is providing the `--from` option, probably they know what they're doing.

We can go back to square one on this subject and make it mandatory that the `--from` option is supplied. We decided not to do for a couple of reasons:

- It's not ergonomic or common in the community to provide the version you're upgrading from.
- It conveys the idea that you can upgrade from whatever older version (although it'd be heavily mitigated by the warning message).

The benefit would be that users will forcefully be more conscious of what they're doing, but not sure if it outweighs the cons.

Reply:
You're right... Let's keep it like this for now, it should be easy enough to change it in the future without breaking anything, if needed.